### ISSUE TITLE
How to Contribute?

### ISSUE THREAD

**Issue by araspik** [tonsky/FiraCode#1066]
I love Fira Code, but I see that you're completely swamped under open issues. How can I help? Is there a particular setup you use to edit the font? Any prior reading / research to understand the whole system?

**Comment by tonsky** [comment]
Thank you! I appreciate your offer. But I’m afraid contributing requires owning Glyphs.app which is quite expensive. That’s a real bummer but I don’t think there’s any decent open-source workflow for producing fonts we could switch to :(

The only other thing I might need help with is building a catalog of all programming combinations, accepted and rejected, with their respective programming languages and meaning. I started this work a while back at https://docs.google.com/spreadsheets/d/1T19dtqd5pqzmUX3epbaGwuVvPIiReCwsd_XDfVhTqW0/edit#gid=0. This might help eventually when we add language tags and language-specific ligatures #76. If you’re interested I can give you edit permissions there.

**Comment by RubenVerg** [comment]
I could help too, if it’s needed. 😊

________________________________

**Comment by araspik** [comment]
> Thank you! I appreciate your offer. But I’m afraid contributing requires owning Glyphs.app which is quite expensive. That’s a real bummer but I don’t think there’s any decent open-source workflow for producing fonts we could switch to :(

Yeah, that sucks. I'll look for some less costly workflows myself, maybe something will turn up.

> The only other thing I might need help with is building a catalog of all programming combinations, accepted and rejected, with their respective programming languages and meaning. I started this work a while back at https://docs.google.com/spreadsheets/d/1T19dtqd5pqzmUX3epbaGwuVvPIiReCwsd_XDfVhTqW0/edit#gid=0. This might help eventually when we add language tags and language-specific ligatures #76. If you’re interested I can give you edit permissions there.

I would rather work without G00gl3, perhaps you could put it (as a text file) in a GitHub gist and give me write access?

**Comment by tonsky** [comment]
> I would rather work without G00gl3, perhaps you could put it (as a text file) in a GitHub gist and give me write access?

It’s not convenient to edit text file with multiple columns.

> Edit: A quick search found FontForge. How does it compare to Glyphs.app?

No idea.

**Comment by RubenVerg** [comment]
suggesting to use a MarkDown gist for the table

**Comment by araspik** [comment]
> suggesting to use a MarkDown gist for the table

I think he meant that tables are difficult in any WYSIWYG-format text file.

> It’s not convenient to edit text file with multiple columns.

It doesn't need to be in a table format. I was thinking something along the lines of
```
:>
 - Used in: Elm, Mathematica
 - Meaning: Update.Extra.andThen RuleDelayed
 - Conflicts: 
 - History:
    - Version Added: 1.205
 - Issues: #547 
...
```
JSON or YAML or similar could also be used.

> > Edit: A quick search found FontForge. How does it compare to Glyphs.app?
> 
> No idea.

I haven't used Glyphs.app, so I can't compare them. Can you try Fontforge out and see whether it works as an alternate workflow?

**Comment by RubenVerg** [comment]
yup

________________________________

**Comment by tonsky** [comment]
> It doesn't need to be in a table format. I was thinking something along the lines of

Ok let’s try that.

> I haven't used Glyphs.app, so I can't compare them. Can you try Fontforge out and see whether it works as an alternate workflow?

I have tried. I think it lacks multi-master editing, it sigfaulted on an attempt on interpalation, and UI is horrible. It’s a no-go.

**Comment by aaronbell** [comment]
@tonsky at the moment it is difficult for me to take on non-paying work, but if there are small tasks that you’d like help with, feel free to ask—I can potentially take some font work off your plate :)

**Comment by tonsky** [comment]
Thanks @aaronbell. That would be fantastic.

One task I have is moving font building to Github Actions. The scripts are here:

https://github.com/tonsky/FiraCode/blob/master/script/build

The environment is defined here (for macOS):

https://github.com/tonsky/FiraCode/blob/master/script/bootstrap

Would be great to being able to automate this

**Comment by aaronbell** [comment]
@tonsky

Embarrassingly, Github Actions are the one thing that I'm not really familiar with. I do local python work for Cascadia, but haven't been able to figure out the Github Action stuff.

**Comment by tonsky** [comment]
Well, like any CI, it’s all about making it work on someone else’s fresh machine, more or less

**Comment by mfocko** [comment]
could have look into it, seems like debian/ubuntu has all dependencies you need
